### ISSUE TITLE
StdOutObserver/StdErrObserver don't receive data because of a clone

### ISSUE THREAD

**Issue by riesentoaster** [AFLplusplus/LibAFL#2266]
Handling of `StdOutObserver` and `StdErrObserver` was recently moved from `Observer` to `CommandExecutor`, which is much cleaner. However, it introduced an issue that didn't get tested I believe:

`CommandConfigurator` now has the following methods:

```rust
/// Get the stdout
    fn stdout_observer(&self) -> Option<&StdOutObserver> {
        None
    }
    /// Get the mut stdout
    fn stdout_observer_mut(&mut self) -> Option<&mut StdOutObserver> {
        None
    }

    /// Get the stderr
    fn stderr_observer(&self) -> Option<&StdErrObserver> {
        None
    }
    /// Get the mut stderr
    fn stderr_observer_mut(&mut self) -> Option<&mut StdErrObserver> {
        None
    }
```

This requires the `CommandConfigurator` to keep a mutable reference to the `StdXObserver` or a `Handle` to those along with a reference to the OT (which obviously isn't possible). And this in turn means the same observer cannot be used in any other way, including passing it to the executor as part of the OT. This will break any `Feedback` that relies on `Handles`, since the observer is not queueable on the OT.

`StdCommandConfigurator` "solves" this cloning the observer, but this doesn't work because then the clone in the configurator is fed the information, while the other clone (in the OT, queued from the feedbacks) is asked for the result, and will never have those.

I'm not sure what the best solution would be here.

@tokatoka made the change in #2167, so maybe you have a solution?

**Comment by riesentoaster** [comment]
@tokatoka Did you have a chance to take a look at this? Or anyone else have a suggestion?

**Comment by tokatoka** [comment]
How about #2301 
The problem is the cloning on the observer, so i stopped cloning the observer itself

**Comment by tokatoka** [comment]
I can't test if it really works so let me know it fixes the problem

**Comment by riesentoaster** [comment]
Closed by #2301 and #2311.
